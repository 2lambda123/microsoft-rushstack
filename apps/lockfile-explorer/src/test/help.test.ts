// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { execSync } from 'child_process';

describe('CLI Tool Tests', () => {
  it('should display help information with --help', () => {
    const output = execSync('node lib/start.js --help').toString();
    expect(output).toMatchSnapshot();
  });
});
