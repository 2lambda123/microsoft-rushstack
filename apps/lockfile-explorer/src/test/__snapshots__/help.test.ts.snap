// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Tool Tests should display help for "lockfile-explorer --help" 1`] = `
"usage: lockfile-explorer [-h] [-d] [--subspace SUBSPACE_NAME] <command> ...

lockfile-lint is a tool for linting lockfiles.

Positional arguments:
  <command>
    start               Start the application

Optional arguments:
  -h, --help            Show this help message and exit.
  -d, --debug           Show the full call stack if an error occurs while 
                        executing the tool
  --subspace SUBSPACE_NAME
                        Specifies an individual Rush subspace to check.

[1mFor detailed help about a specific command, use: lockfile-explorer 
<command> -h[22m
"
`;

exports[`CLI Tool Tests should display help for "lockfile-lint --help" 1`] = `
"[33mThe experiment \\"phasedCommands\\" has graduated to a standard feature. Remove this experiment from \\"/workspaces/rushstack/common/config/rush/experiments.json\\".[39m
usage: lockfile-lint [-h] <command> ...

lockfile-lint is a tool for linting lockfiles.

Positional arguments:
  <command>
    init      Create lockfile-lint.json config file
    lint      Check if the specified package has a inconsistent package 
              versions in target project

Optional arguments:
  -h, --help  Show this help message and exit.

[1mFor detailed help about a specific command, use: lockfile-lint <command> 
-h[22m
"
`;
